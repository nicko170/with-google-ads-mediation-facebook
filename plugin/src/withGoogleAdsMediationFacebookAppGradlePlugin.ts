import {ConfigPlugin, WarningAggregator, withAppBuildGradle} from '@expo/config-plugins';

const withGoogleAdsMediationFacebookAppGradlePlugin: ConfigPlugin = config => {
    return withAppBuildGradle(config, ({modResults, ...config}) => {
        if (modResults.language !== 'groovy') {
            WarningAggregator.addWarningAndroid(
                'withGoogleMobileAdsMediationFacebook',
                `Cannot automatically configure app build.gradle if it's not groovy`,
            );
            return {modResults, ...config};
        }

        modResults.contents = addFacebookImplementation(modResults.contents);
        return {modResults, ...config};
    });
};

const addFacebookImplementation = (projectBuildGradle: string) => {
    if (projectBuildGradle.includes('com.google.ads.mediation:facebook')) return projectBuildGradle;

    return projectBuildGradle.replace(
        /dependencies\s?{/,
        `dependencies {
        implementation 'com.google.ads.mediation:facebook:6.8.0.0'
    `,
    );
};

export {addFacebookImplementation};
export default withGoogleAdsMediationFacebookAppGradlePlugin;