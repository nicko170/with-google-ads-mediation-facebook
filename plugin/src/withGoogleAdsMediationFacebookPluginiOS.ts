import {ConfigPlugin, withDangerousMod} from '@expo/config-plugins';
import {mergeContents, MergeResults} from "@expo/config-plugins/build/utils/generateCode";
import fs from 'fs/promises';
import path from 'path';

const withGoogleAdsMediationFacebookPluginiOS: ConfigPlugin = config => {
    return withDangerousMod(config, [
        'ios',
        async config => {
            const filePath = path.join(config.modRequest.platformProjectRoot, 'Podfile');
            const contents = await fs.readFile(filePath, 'utf-8');
            let results: MergeResults;
            try {
                results = addGoogleMobileAdsMediationFacebookSupport(contents)
            } catch (e) {
                if (e.code == 'ERR_NO_MATCH') {
                    throw new Error(
                        `Cannot add GoogleMobileAdsMediationFacebook to the project's ios/Podfile because it's malformed. Please report this with a copy of your project Podfile.`
                    );
                }
                throw e;
            }

            if (results.didMerge) {
                await fs.writeFile(filePath, results.contents);
            }

            return config
        }
    ]);
};


const addGoogleMobileAdsMediationFacebookSupport = (src: string): MergeResults => {
    return mergeContents({
        tag: 'google-mobile-ads-mediation-facebook',
        src,
        newSrc: `  pod 'GoogleMobileAdsMediationFacebook'`,
        anchor: /use_native_modules/,
        offset: 0,
        comment: '#',
    });
}


export {addGoogleMobileAdsMediationFacebookSupport};
export default withGoogleAdsMediationFacebookPluginiOS;
