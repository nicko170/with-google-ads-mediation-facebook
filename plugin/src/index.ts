import { ConfigPlugin, withPlugins } from '@expo/config-plugins';

import withGoogleAdsMediationFacebookPluginiOS from './withGoogleAdsMediationFacebookPluginiOS';

const withGoogleIDFASupport: ConfigPlugin = (config) => {
  return withPlugins(config, [withGoogleAdsMediationFacebookPluginiOS]);
};

export default withGoogleIDFASupport;
