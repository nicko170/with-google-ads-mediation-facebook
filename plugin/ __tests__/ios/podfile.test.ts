import * as fs from 'fs/promises';
import * as path from 'path';

import * as Plugin from '../../src/withGoogleAdsMediationFacebookPluginiOS';

describe('[iOS] Podfile Test', () => {
  it('adds GoogleMobileAdsMediationFacebook to iOS Podfile', async () => {
    let podfile = await fs.readFile(path.resolve(__dirname, '../fixtures/Podfile'), {
      encoding: 'utf-8',
    });

    const podfileWithGoogleIDFASupport = Plugin.addGoogleMobileAdsMediationFacebookSupport(podfile)

    expect(podfileWithGoogleIDFASupport).toMatchSnapshot();
  });
});
