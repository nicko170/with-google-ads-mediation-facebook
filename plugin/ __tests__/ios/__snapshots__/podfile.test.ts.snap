// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[iOS] Podfile Test adds GoogleMobileAdsMediationFacebook to iOS Podfile 1`] = `
Object {
  "contents": "require File.join(File.dirname(\`node --print \\"require.resolve('expo/package.json')\\"\`), \\"scripts/autolinking\\")
require File.join(File.dirname(\`node --print \\"require.resolve('react-native/package.json')\\"\`), \\"scripts/react_native_pods\\")
require File.join(File.dirname(\`node --print \\"require.resolve('@react-native-community/cli-platform-ios/package.json')\\"\`), \\"native_modules\\")

platform :ios, '12.0'

require 'json'
podfile_properties = JSON.parse(File.read('./Podfile.properties.json')) rescue {}

target 'RFSPager' do
  use_expo_modules!
# @generated begin google-mobile-ads-mediation-facebook - expo prebuild (DO NOT MODIFY) sync-66d41c2d1c8e14f71bc650a9b46279f3c0bcf5bd
  pod 'GoogleMobileAdsMediationFacebook'
# @generated end google-mobile-ads-mediation-facebook
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => podfile_properties['expo.jsEngine'] == 'hermes'
  )

  # Uncomment to opt-in to using Flipper
  #
  # if !ENV['CI']
  #   use_flipper!('Flipper' => '0.75.1', 'Flipper-Folly' => '2.5.3', 'Flipper-RSocket' => '1.3.1')
  # end

  post_install do |installer|
    react_native_post_install(installer)

    # Workaround \`Cycle inside FBReactNativeSpec\` error for react-native 0.64
    # Reference: https://github.com/software-mansion/react-native-screens/issues/842#issuecomment-812543933
    installer.pods_project.targets.each do |target|
      if (target.name&.eql?('FBReactNativeSpec'))
        target.build_phases.each do |build_phase|
          if (build_phase.respond_to?(:name) && build_phase.name.eql?('[CP-User] Generate Specs'))
            target.build_phases.move(build_phase, 0)
          end
        end
      end
    end
  end

end
",
  "didClear": false,
  "didMerge": true,
}
`;
